src/bin/cargo/cli.rs

```diff
@@ -514,7 +513,19 @@ impl GlobalArgs {
     }
 }
 
-pub fn cli() -> Command {
+pub fn cli(gctx: &GlobalContext) -> Command {
+    // Don't let config errors get in the way of parsing arguments
+    let term = gctx.get::<TermConfig>("term").unwrap_or_default();
```

Comment:
The issue is that we sidestep without calling `GlobalContext::merge_cli_args`, which `GlobalContext::configure` does. 

There might also be a discrepancy when we call `reload_cwd()`, but that is minimal as the code path eventually calls `GlobalContext::configure`.

Reply:
I'm not too sure what the concern is here.

Is it that we aren't respecting `--color` which you raised elsewhere and you are just calling out the relevant code?

Or is it that we aren't respecting the config from `-C` to control the colors from CLI parsing?  If so, that is a similar problem to `--color` support (and more difficult / brittle to support)

Reply:
I was just making sure we're aware of this and also #9012 is not completely resolved. Should we reword #9012 to track `--config` and `--color` or open a new one?

I am good with merge this PR as-is btw. Feel free to `r=weihanglo`.